src/odemis/driver/ephemeron.py

```diff
@@ -0,0 +1,747 @@
+    Argument(Name="delay",
+             DataType=NodeId(ObjectIds.Float),
+             ValueRank=-1,
+             ArrayDimensions=[],
+             Description=LocalizedText("Delay time (ms)"))
+]
+
+SCANTIME_ARGS = [
+    Argument(Name="dwell_time",
+             DataType=NodeId(ObjectIds.Int64),
+             ValueRank=-1, ArrayDimensions=[],
+             Description=LocalizedText("Dwell Time (ms)")),
+    Argument(Name="points_fast",
+             DataType=NodeId(ObjectIds.Int64),
+             ValueRank=-1,
+             ArrayDimensions=[],
+             Description=LocalizedText("Points Fast")),
+    Argument(Name="points_slow",
+             DataType=NodeId(ObjectIds.Int64),
+             ValueRank=-1,
+             ArrayDimensions=[],
+             Description=LocalizedText("Points Slow"))
+]
+
+STATE_ARGS = [
+    Argument(Name="new_state",
+             DataType=NodeId(ObjectIds.String),
+             ValueRank=-1,
+             ArrayDimensions=[],
+             Description=LocalizedText("Requested State"))
+]
+
+
+class MightyEBIC(Detector):
+    def __init__(self, name: str, role: str, channel: int, url: str, namespace: str, **kwargs):
+        """
+        Initialise the EBIC controller
+        :param name (str): The name of the device configured through the configuration file
+        :param role (str): The role of the device configured through the configuration file
+        :param channel (int): The channel the device should use for hw triggering.
+        :param url (str): The url address to use with the OPC UA protocol
+        :param namespace (str): The object name identifier to use for the OPC UA protocol
+        """
+        super().__init__(name, role, **kwargs)
+
+        self._channel = channel
+        self._url = url
+        self._namespace = namespace
+
+        # server_sim is only used for simulation of the opcServer if it is non-existent
+        self._opc_server_sim: Optional[MightyEBICSimulator] = None
+        self._opc_client: Optional[UaClient] = None
+        self._error_msg: Optional[str] = None
+        self._t_simserver: Optional[threading.Thread] = None
+        # hardcoded now, eventually -> get resolution from the controller/detector
+        self._ebeam_res = ((1, 1), (4096, 4096))
+        self._server_exception: Optional[Exception] = None
+
+        if "localhost" in self._url:
+            # if the device should be simulated, start a simulated server first
+            self._opc_server_sim = MightyEBICSimulator(self._url, self._namespace, self)
+            self._t_simserver = threading.Thread(target=self._start_opc_simserver)
+            self._t_simserver.start()
+            # start the simulated server threaded but wait for it to be ready (running)
+            while not self._opc_server_sim.ready:
+                time.sleep(0.1)
+                if self._server_exception:
+                    raise HwError(self._server_exception.args[0])
```

Comment:
It should be possible to move all of this (and the start_opc_simserver method) to the simulator (init). Then, only provide a .terminate() function on the simulator to stop it (ie, do what the .terminate() currently does).